commands/list.rb

```diff
@@ -0,0 +1,49 @@
+require 'fileutils'
+require_relative '../lib/color'
+require_relative '../lib/const'
+require_relative '../lib/package'
+
+class Command
+  def self.list(available, installed, compatible, incompatible, verbose)
+    device_json = JSON.load_file('/usr/local/etc/crew/device.json', symbolize_names: true)
+    installed_packages = {}
+    device_json[:installed_packages].each do |package|
+      installed_packages[package[:name]] = package[:version]
+    end
+
+    if available
+      Dir["#{CREW_PACKAGES_PATH}/*.rb"].each do |filename|
+        pkg_name = File.basename(filename, '.rb')
+        next if installed_packages.key?(pkg_name)
+        pkg = Package.load_package(filename)
+        puts pkg_name if pkg.compatible?
+      end
+    elsif installed
```

Comment:
Use `case ... when` instead of `if ... elsif`?

Reply:
Because they're all passed as separate arguments (in order to avoid a second case statement in `bin/crew`), we cant use a `case` statement.

Reply:
You get rid of the if statement altogether if you simply pass args and then make the decision in the function instead.